### PR TITLE
HSEARCH-1640 Probe boolean field projection and internal representation

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-1640

### CODE REVIEW

File: integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		IndexAccessors(IndexSchemaElement root) {
+			string = root.field( "string", f -> f.asString().projectable( Projectable.YES ) ).createAccessor();
+			text = root.field( "text", f -> f.asString().analyzer( DefaultAnalysisDefinitions.ANALYZER_STANDARD.name ).projectable( Projectable.YES ) ).createAccessor();
+			integer = root.field( "integer", f -> f.asInteger().projectable( Projectable.YES ) ).createAccessor();
+			longNumber = root.field( "longNumber", f -> f.asLong().projectable( Projectable.YES ) ).createAccessor();
+			bool = root.field( "bool", f -> f.asBoolean().projectable( Projectable.YES ) ).createAccessor();
+
+			// Emulates what we do in LuceneBooleanFieldCodec:
+			// the external form is the Boolean,
+			// BUT we threat it as **Integer** for comparison operation (range, sort)
```
Comment (yrodiere) [inline integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java]: "threat" => "treat"

File: integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			// and we store it as **Integer** as well.
+			nativeBool = root.field( "nativeBool", f ->
```
Comment (yrodiere) [inline integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java]: Not sure why you decided to test this? You're basically testing code that you defined in your tests, so I doubt it is useful.

File: integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+						ctx -> this.indexAccessors = new IndexAccessors( ctx.getSchemaElement() ),
+						indexManager -> this.indexManager = indexManager
+				)
+				.setup();
+
+		initData();
+	}
+
+	@Test
+	public void verifyProjectionsOnDifferentTypes_usingStandardAPI() {
```
Comment (yrodiere) [inline integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java]: This is already tested extensively in the backend TCK. You can remove this test, it's not worth the maintenance cost.

File: integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java (hunk 2 above)
Comment (fax4ever) [reply integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/fieldtype/LuceneFieldContentIT.java]: Your're right, I'll remove it

**Comment by fax4ever** [comment]
@yrodiere Removed native boolean field and standard projections.

**Comment by yrodiere** [comment]
@fax4ever Thanks. I added a commit to avoid a dependency between test classes. Let's merge once CI goes green.

**Comment by yrodiere** [comment]
Merged, thanks!

**Comment by fax4ever** [comment]
Thanks

**Pull request merged**